widget/embedded/src/pages/Home.tsx

```diff
@@ -189,20 +205,68 @@ export function Home() {
           image: swap.to.blockchainLogo,
         },
         price: {
-          value: outputAmount
-            ? numberToString(new BigNumber(outputAmount))
-            : numberToString(new BigNumber(0)),
+          value: swap.toAmount,
         },
       },
+      alerts:
+        swap.swapperId === swapHasError?.swapperId ? (
+          <FooterStepAlarm>
+            <Alert
+              type="error"
+              title={recommendation}
+              footer={
+                <FooterAlert>
+                  <Typography size="xsmall" variant="body" color="neutral900">
+                    {fromAmountRangeError}
+                  </Typography>
+                  <Divider direction="horizontal" size={8} />
+                  <Typography size="xsmall" variant="body" color="neutral900">
+                    |
+                  </Typography>
+                  <Divider direction="horizontal" size={8} />
+                  <Typography size="xsmall" variant="body" color="neutral900">
+                    {`Yours: ${numberToString(
+                      swapHasError?.fromAmount || null
+                    )} ${swap?.from.symbol}`}
+                  </Typography>
+                </FooterAlert>
+              }
+            />
+          </FooterStepAlarm>
+        ) : undefined,
     }));
   };
 
   const totalFeeInUsd = getTotalFeeInUsd(bestRoute, tokens);
-  const highFee = hasHighFee(totalFeeInUsd);
+  // const highFee = hasHighFee(totalFeeInUsd);
+
+  const bestRouteData = getFormatedBestRoute(bestRoute);
 
   return (
     <Layout
       hasFooter
+      action={
+        <Button
+          type="primary"
+          size="large"
+          disabled={swapButtonState.disabled}
+          prefix={
+            !swapButtonState.disabled &&
+            swapButtonState.hasWarning && <WarningIcon />
+          }
+          fullWidth
+          onClick={() => {
+            if (swapButtonState.title === 'Connect Wallet') {
```

Comment:
Also please create a map for swap button state' messages and read from there if it is possible.